src/commands/Games/higherLower.ts

```diff
@@ -0,0 +1,267 @@
+			for (const emoji of game.emojis) {
+				await game.response.react(emoji);
+			}
+
+			const emoji = await this.awaitReactionAction(game);
+			if (emoji === null) break;
+
+			// Main game logic (at last)
+			const oldNum = game.number;
+			game.number = this.random(game.number);
+
+			switch (emoji) {
+				case HigherLowerReactions.Higher: {
+					game.running = await (game.number > oldNum ? this.win(game, message) : this.loss(game, message));
+					break;
+				}
+				case HigherLowerReactions.Lower: {
+					game.running = await (game.number < oldNum ? this.win(game, message) : this.loss(game, message));
+					break;
+				}
+				case HigherLowerReactions.Cashout: {
+					await this.end(game, message, true);
+					game.running = false;
+					break;
+				}
+			}
+
+			if (game.running) game.turn++;
+		}
+	}
+
+	private async win(game: HigherLowerGameData, message: KlasaMessage) {
+		const { language } = message;
+		await game.response.reactions.removeAll();
+
+		// TODO(Quantum): Implement Winning event for InfluxDB
+		const { TITLE, DESCRIPTION, FOOTER } = message.language.tget('COMMAND_HIGHERLOWER_WIN');
+		await game.response.edit(null, new MessageEmbed()
+			.setColor(getColor(message))
+			.setTitle(TITLE)
+			.setDescription(DESCRIPTION(this.calculateWinnings(game.wager, game.turn), game.number))
+			.setFooter(FOOTER));
+
+		// Ask the user whether they want to continue or cashout
+		game.emojis = this.kWinReactionArray;
+		for (const emoji of game.emojis) {
+			await game.response.react(emoji);
+		}
+
+		const reaction = await this.awaitReactionAction(game);
+		await game.response.reactions.removeAll();
+
+		const action = reaction === null
+			? EndingAction.Timeout
+			: reaction === HigherLowerReactions.Ok
+				? EndingAction.Play
+				: EndingAction.Stop;
+
+		// Decide whether we timeout, stop, or continue
+		switch (action) {
+			case EndingAction.Timeout:
+				await game.response.edit(language.tget('COMMAND_HIGHERLOWER_TIMEOUT'), { embed: null });
+			case EndingAction.Stop:
+				await this.end(game, message, true);
+				break;
+			case EndingAction.Play:
+				await game.response.edit(language.tget('COMMAND_HIGHERLOWER_NEWROUND'), { embed: null });
+				break;
+		}
+		return action === EndingAction.Play;
+	}
+
+	private async loss(game: HigherLowerGameData, message: KlasaMessage) {
+		// TODO (Quantum): Implement losing event to InfluxDB
+		const { number, wager, turn, response } = game;
+		await response.reactions.removeAll();
+		let losses = wager;
+
+		// There's a 0.1% chance that a user would lose now only the wager, but also what they would've won. Muahaha
+		if ((Math.random()) < 0.001) {
+			losses += this.calculateWinnings(wager, turn - 1);
+			await message.author.settings.decrease(UserSettings.Money, losses);
+		}
+
+		const { TITLE, DESCRIPTION, FOOTER } = message.language.tget('COMMAND_HIGHERLOWER_LOSE');
+		await game.response.edit(null, new MessageEmbed()
+			.setColor(getColor(message))
```

Comment:
`getColor(message)` should be moved to a private property so it doesn't have to be resolved 4x. Though I suppose the message is not accessible in the constructor, is it? In that case I suppose set it at the start of the run() and reference it in the other private methods where it's being used.

Reply:
Can be moved to the game context, no problem.